deepmd/dpmodel/fitting/general_fitting.py

```diff
@@ -364,6 +364,11 @@ def _call_common(
 
         """
         xp = array_api_compat.array_namespace(descriptor, atype)
+        descriptor = xp.astype(descriptor, get_xp_precision(xp, self.precision))
```

Comment:
it seems that gr, g2 and h2 are not cast.

Reply:
gr, g2 and h2 are not used below. I don't know why they are input.